physical/gcs/gcs_test.go

```diff
@@ -84,5 +84,183 @@ func TestGCSBackend(t *testing.T) {
 
 	physical.ExerciseBackend(t, b)
 	physical.ExerciseBackend_ListPrefix(t, b)
+}
+
+func TestGCSHABackend(t *testing.T) {
+	credentialsFile := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+
+	// projectID is only required for creating a bucket for this test
+	projectID := os.Getenv("GOOGLE_PROJECT_ID")
+
+	if credentialsFile == "" || projectID == "" {
+		t.SkipNow()
+	}
+
+	client, err := storage.NewClient(
+		context.Background(),
+		option.WithServiceAccountFile(credentialsFile),
+	)
```

Comment:
Can we use the new user-agent helper here to track usage?

```go
option.WithUserAgent(useragent.String())
```

It might not be possible to do in a BC way since this integration already exists, but using `WithServiceAccountFile` makes it challenging to use this integration with application default credentials. If you just don't supply the option, user's can specify the `GOOGLE_APPLICATION_CREDENTIALS` envvar instead, falling back to app default creds.